google/cloud/spanner_dbapi/cursor.py

```diff
@@ -309,19 +315,26 @@ def executemany(self, operation, seq_of_params):
         self._result_set = None
         self._row_count = _UNSET_COUNT
 
-        class_ = parse_utils.classify_stmt(operation)
-        if class_ == parse_utils.STMT_DDL:
+        parsed_statement = parse_utils.classify_stmt(operation)
+        if parsed_statement.statement_type == StatementType.DDL:
             raise ProgrammingError(
                 "Executing DDL statements with executemany() method is not allowed."
             )
 
         # For every operation, we've got to ensure that any prior DDL
         # statements were run.
         self.connection.run_prior_DDL_statements()
+        if parsed_statement.statement_type == StatementType.CLIENT_SIDE:
+            raise ProgrammingError(
+                "Executing ClientSide statements with executemany() method is not allowed."
```

Comment:
Do we explain somewhere to customers what a "ClientSide" statement is?

Is this necessarily a requirement?  Could we just split the batch on the client-side statement, then run everything before as a batch, then do the client-side statement, then run everything after as a batch?  That might be out of scope for this PR, though; just asking if it makes sense in general.

Reply:
Let me give a better error message.

Sorry might be I have not understood the comment completely. Do you mean if multiple statements are passed to executemany() (please remember its not possible) and one of them is client side statement?

Reply:
Changed the error message